src/ucp/core/ucp_context.c

```diff
@@ -639,6 +642,19 @@ static ucs_status_t ucp_fill_resources(ucp_context_h context,
          * client-server connection establishment via sockaddr, don't use it */
         if ((num_tl_resources > 0) ||
             (context->tl_mds[md_index].attr.cap.flags & UCT_MD_FLAG_SOCKADDR)) {
+            /*List of memory type MDs*/
+            if (context->tl_mds[md_index].attr.cap.mem_type != UCT_MD_MEM_TYPE_HOST) {
+                context->mem_type_tl_mds = ucs_realloc(context->mem_type_tl_mds,
+                                                       sizeof(*context->mem_type_tl_mds) *
+                                                       (1 + context->num_mem_type_mds),
+                                                       "ucp_mem_type_tl_mds");
+                if (context->mem_type_tl_mds == NULL) {
+                    status = UCS_ERR_NO_MEMORY;
+                    goto err_free_context_resources;
+                }
+                context->mem_type_tl_mds[context->num_mem_type_mds] = context->tl_mds[md_index];
```

Comment:
maybe save a pointer to the ucp_tl_md instead of copying it?

Reply:
ok.